imapfilter: initial package for imapfilter v2.6.16

VanStratum opened the pull request:
_Motivation:_  Initial package for imapfilter. Imapfilter allows to apply complex mail filtering rules over IMAP using lua as a scripting language.
_Linked issues:_ 

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully
The package works properly, but for some reasons unknown to me, the executable files are not linked properly to /usr/local.

VanStratum committed:
imapfilter: initial package for imapfilter v2.6.16

hgy59 commented on cross/imapfilter/Makefile:
```diff
@@ -0,0 +1,17 @@
+HOMEPAGE = http://github.com/lefcha/imapfilter
+COMMENT  = An IMAP mail filtering utility using the Lua programming language as a configuration and extension language.
+LICENSE  = MIT
+UNSUPPORTED_ARCHS = ppc853x
```
You can add the following snippet to add support for ppc853x:
```
ifeq ($(findstring $(ARCH),powerpc ppc824x ppc853x ppc854x),$(ARCH))
# PPC_ARCHES except QorIQ
ADDITIONAL_CFLAGS = -DLUA_32BITS
endif
```

hgy59 commented on spk/imapfilter/Makefile:
```diff
@@ -0,0 +1,19 @@
+DESCRIPTION = An IMAP mail filtering utility using the Lua programming language as a configuration and extension language.
+STARTABLE = no
+DISPLAY_NAME = Imapfilter
+BETA = 1
```
As imapfilter it self looks stable I would omit BETA=1

hgy59 reviewed (changes requested):
Thanks for the great job done.
Proposed some small changes.

Since imapfilter is a CLI tool for advanced users I propose to add it to `synocli-net`.
And when adding to synocli-net, move the spk/imapfilter to diyspk/imapfilter.

There is no automatic link creation for the executables to /usr/local/bin. You have to provide a service-setup script for this.

VanStratum committed:
imapfilter: implement changes suggested in #3836

Add service-setup script.
Change name of downloaded tgz.
Fix dependencies of the spk.
Add rule to compile on ppc platforms.
Remove beta status.
Move to diyspk.

VanStratum commented:
Thank you for the feedback.
	I added the requested changes, but before adding the package to synocli-net, I would like to know how I should increment the the version number and the revision, and how the changelog should be written, because the history of it doesn't seems consistent.

VanStratum commented:
If I understand correctly the way the changelog and the revision number works,
the revision number should be 5 and the changelog
```
CHANGELOG = "5. Add imapfilter
4. Add rsync
3. Build sshfs 2.10 with fuse updated to v2.9.9
2. Add fritzctl
1. First package release"
```
Should I commit this changelog and "correct" the revision number, or am I misunderstanding something?

ymartin59 commented on diyspk/imapfilter/Makefile:
```diff
@@ -0,0 +1,19 @@
+HOMEPAGE = http://github.com/lefcha/imapfilter
+LICENSE  = MIT
+
+INSTALL_PREFIX = /usr/local
```
Please discard this redundant line.

ymartin59 commented on diyspk/imapfilter/src/service-setup.sh:
```diff
@@ -0,0 +1,11 @@
+service_postinst ()
+{
+    ln -s ${SYNOPKG_PKGDEST}/bin/imapfilter /usr/local/bin/imapfilter
+    ln -s ${SYNOPKG_PKGDEST}/share/imapfilter /usr/local/share/
```
I would not link that folder. I propose to build tool replacing default "share" location to `/var/packages/$(PKG_NAME)/target/share/` - with variable to work either with diyspk/imapfilter or spk/synocli-net

ymartin59 reviewed (changes requested):
@hgy59 In case we get Meson to cross-compile for fuse in synocli-net

VanStratum commented:
Thank you for your review and your suggestions. I am currently drowned
in work and unable to work on this right now, but I will apply your
changes as soon as possible.

hgy59 committed:
update imapfilter to v2.7.5

- adjust dependencies (cross/pcre replaced by cross/pcre2)
- include sample lua scripts
- use generic link creation for bin file and share folder

hgy59 committed:
add IMAPFilter to synocli-net

hgy59 commented:
@th0ma7 the meson build does not work in the spksrc build environment.
It tries to run cross-compiled sanitycheckc.exe:
```
meson.build:1:0: ERROR: Could not invoke sanity test executable: [Errno 8] Exec format error: '/spksrc/spk/synocli-file/work-armv7-6.1/glib-2.62.6/builddir/meson-private/sanitycheckc.exe'.
```

th0ma7 commented:
> @th0ma7 the meson build (#4307) does not work in the spksrc build environment.
> It tries to run cross-compiled sanitycheckc.exe:
> 
> ```
> meson.build:1:0: ERROR: Could not invoke sanity test executable: [Errno 8] Exec format error: '/spksrc/spk/synocli-file/work-armv7-6.1/glib-2.62.6/builddir/meson-private/sanitycheckc.exe'.
> ```

I suspect this might be post #4307 as last I checked it passed all green on github-action...
My PC is offline for 1-2 more days but I'll have a look into it once back online.
In the meantime, does this PR build directly on top of https://github.com/SynoCommunity/spksrc/commit/5af14c588093827696e382a2476e9e3ee63c6bc7 ?

th0ma7 commented:
@hgy59 Found it!   `ARCHES` to `ARCHS` conversion typo in `spksrc.cross-meson-env.mk`, thus in turn not providing the cross-build environment to `meson` :
```
ifeq ($(findstring $(ARCH),$(ARMv5_ARCHS)),$(ARCH))
  CONFIGURE_ARGS += --cross-file $(MESON_CFG_DIR)/armv5.cfg
endif
ifeq ($(findstring $(ARCH),$(ARMv7_ARCHES)),$(ARCH))
  CONFIGURE_ARGS += --cross-file $(MESON_CFG_DIR)/armv7.cfg
endif
ifeq ($(findstring $(ARCH),$(ARMv7L_ARCHS)),$(ARCH))
  CONFIGURE_ARGS += --cross-file $(MESON_CFG_DIR)/armv7l.cfg
endif
```

If you don't mind I leave it up to you to push a fix for it.

hgy59 committed:
use package specific share as DSM resource linker does not support /usr/local/share

hgy59 commented:
sad to say that `SPK_USR_LOCAL_LINKS` definitly does not support the creation of linked folders in `/usr/local/share` (#4578)

hgy59 commented:
@VanStratum _Good things come to those who wait._
Just updated this long waiting request, but now it runs even on DSM 7.

hgy59 committed:
fix lua for arch-ppc853x

hgy59 committed:
revert changes in cross/lua/Makefile

hgy59 committed:
revert integration in synocli-net/Makefile

hgy59 committed:
Merge branch 'master' of https://github.com/SynoCommunity/spksrc into imapfilter

hgy59 committed:
imapfilter: update to v2.8.3

- update imapfilter from v2.7.5 to v2.8.3
- update openssl to openssl3
- configure ssl paths for DSM

hgy59 committed:
synocli-net: add imapfilter again

hgy59 commented:
@VanStratum this is a really old PR.
It is now updated with upstream repository and builds again.
Are you still interested in imapfilter on Synology devices?

hgy59 committed:
Merge branch 'master' of https://github.com/SynoCommunity/spksrc into imapfilter

hgy59 commented:
Merge to prepare for updating the synocli-net package
